modules/clients/src/test/java/org/apache/ignite/jdbc/JdbcErrorsAbstractSelfTest.java

```diff
@@ -742,6 +743,61 @@ public void testDdlWrongSyntax() throws SQLException {
             "Failed to parse query. Syntax error in SQL statement \"ALTER TABLE TEST DROP COLUMN [*]");
     }
 
+    /**
+     * Checks execution DML request on read-only cluster error code and message.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testUpdatesRejectedInReadOnlyMode() throws Exception {
+        try (Connection conn = getConnection()) {
+            try (Statement statement = conn.createStatement()) {
+                statement.executeUpdate("CREATE TABLE TEST_READ_ONLY (ID LONG PRIMARY KEY, VAL LONG)");
+            }
+        }
+
+        grid(0).cluster().state(ClusterState.ACTIVE_READ_ONLY);
+
+        try {
+            checkErrorState((conn) -> {
```

Comment:
i found that **create table** still possible in ro mode:
`        grid(0).cluster().state(ClusterState.ACTIVE_READ_ONLY);

        try {
            try (Connection conn = getConnection()) {
                try (Statement statement = conn.createStatement()) {
                    statement.executeUpdate("CREATE TABLE TEST_READ_ONLY (ID LONG PRIMARY KEY, VAL LONG)");
                }
            }
        }
        finally {
            grid(0).cluster().state(ClusterState.ACTIVE);
        }`

Reply:
This ticket is about a cluster state change API, not about cluster read-only mode. 
The problem will be fixed in https://ggsystems.atlassian.net/browse/GG-29471 ticket.